staging/src/k8s.io/client-go/tools/clientcmd/validation.go

```diff
@@ -207,35 +207,39 @@ func validateAuthInfo(authInfoName string, authInfo clientcmdapi.AuthInfo) []err
 		methods = append(methods, "basicAuth")
 	}
 
-	if len(authInfo.ClientCertificate) != 0 || len(authInfo.ClientCertificateData) != 0 {
+	if len(authInfo.ClientCertificate) != 0 {
 		// Make sure cert data and file aren't both specified
-		if len(authInfo.ClientCertificate) != 0 && len(authInfo.ClientCertificateData) != 0 {
+		if len(authInfo.ClientCertificateData) != 0 {
 			validationErrors = append(validationErrors, fmt.Errorf("client-cert-data and client-cert are both specified for %v. client-cert-data will override.", authInfoName))
 		}
-		// Make sure key data and file aren't both specified
-		if len(authInfo.ClientKey) != 0 && len(authInfo.ClientKeyData) != 0 {
-			validationErrors = append(validationErrors, fmt.Errorf("client-key-data and client-key are both specified for %v; client-key-data will override", authInfoName))
-		}
-		// Make sure a key is specified
-		if len(authInfo.ClientKey) == 0 && len(authInfo.ClientKeyData) == 0 {
-			validationErrors = append(validationErrors, fmt.Errorf("client-key-data or client-key must be specified for %v to use the clientCert authentication method.", authInfoName))
+
+		clientCertFile, err := os.Open(authInfo.ClientCertificate)
+		defer clientCertFile.Close()
+		if err != nil {
+			validationErrors = append(validationErrors, fmt.Errorf("unable to read client-cert %v for %v due to %v", authInfo.ClientCertificate, authInfoName, err))
 		}
+	}
+	// Make sure a key is specified
+	if len(authInfo.ClientCertificate) == 0 && len(authInfo.ClientCertificateData) == 0 {
```

Comment:
this should not have been unnested... the previous logic checked that if client cert or certdata was specified, then either client key or keydata was also specified. this change makes it an error to not specify a client cert, which is incorrect.